src/cimdemo/EE_geoJSON_ex.py

```diff
@@ -0,0 +1,181 @@
+import geopandas as gp
+from tkinter import filedialog
+from tkinter import *
+from pathlib import Path
+import logging
+import asyncio
+import argparse
+import pydash
+import json
+import os
+
+logging.basicConfig(level=logging.DEBUG)
+logger = logging.getLogger(__name__)
+
+
+def get_path():
+    root = Tk()
+    root.filename = filedialog.askopenfilename(initialdir=Path.home(), title="Select file",
+                                               filetypes=(("jpeg files", "*.geojson"), ("all files", "*.*")))
+    return root.filename
+
+
+def read_mapping(path):
+    # Opening JSON file
+    file = open(path, "r")
+    # returns JSON object as a dictionary
+    return json.loads(file.read())["mappings"]
+
+
+def read_json_file(path):
+    # Opening JSON file
+    with open(path, "r") as f:
+        return json.loads(f.read())
+
+
+def add_list_to_net(l, net):
+    return net
+
+
+class Network:
+
+    def __init__(self):
+        self.path = get_path()
+        self.geojson_file = read_json_file(self.path)
+        self.mapping = read_json_file('ee-mapping.json')
+        self.feeder_name = os.path.basename(self.path)
+        self.gdf = gp.read_file(self.path)
+        self.ns = cim.NetworkService()
+        self.ds = cim.DiagramService()
+
+        self.voltages = self.add_base_voltages()
+
+    def get_cim_class(self, gis_class):
+        if self.mapping.get(gis_class):
+            return self.mapping[gis_class]["cimClass"]
+        else:
+            return None
+
+    def add_diagram(self):
+        diagram = cim.Diagram(diagram_style=cim.DiagramStyle.GEOGRAPHIC)
+        self.ds.add(diagram)
+        return diagram
+
+    def add_location(self, row):
+        loc = cim.Location()
+        for coord in row["geometry"].coords:
+            logger.info("Creating coordinates: " + coord.__str__())
+            loc.add_point(cim.PositionPoint(coord[0], coord[1]))
+            logger.info('Add Location to Network Service')
+            self.ns.add(loc)
+        return loc
+
+    def add_base_voltages(self):
+        self.ns.add(cim.BaseVoltage(mrid='LV', nominal_voltage=415, name='415V'))
+        self.ns.add(cim.BaseVoltage(mrid='Service', nominal_voltage=415, name='415V'))
+        self.ns.add(cim.BaseVoltage(mrid='11kV', nominal_voltage=11000, name='11kV'))
+        self.ns.add(cim.BaseVoltage(mrid='UNKNOWN', nominal_voltage=415, name='415V'))
+
+    def create_equipment(self, row, loc):
+        class_name = self.get_cim_class(row['class'])
+        if class_name is not None:
+            logger.info("Creating CIM Class: " + class_name)
+            class_ = getattr(cim, class_name)
+            eq = class_()
+            logger.info('Creating Equipment:' + ", mRID: " + eq.mrid.__str__())
+            eq.mrid = row["id"]
+            eq.name = row["name"]
+            eq.location = loc
+            if row['operating voltage'] is not None:
+                logger.info('Assigning BaseVoltage: ' + row['operating voltage'].__str__())
```

Comment:
```suggestion
                logger.info(f'Assigning BaseVoltage: {row['operating voltage']})
```
this is a more concise way to do string formatting - called an fstring (note the f at the start), which allows you to interpolate code within `{}` brackets inside the string. Note you also shouldn't have to call `str()`, this will be done for you. 
Feel free to redo your log strings as you see fit, you don't have to do it this way.